src/sentry/models/apitoken.py

```diff
@@ -28,10 +29,98 @@ def default_expiration():
     return timezone.now() + DEFAULT_EXPIRATION
 
 
-def generate_token():
+def generate_token(token_type: AuthTokenType | str | None = AuthTokenType.__empty__) -> str:
```

Comment:
I was running in to some issues with `mypy` on this when I originally just had the type as `AuthTokenType`. 
The particular field on the class is a `CharField` so it's expecting a `str`.

Is there a way around this?

https://github.com/getsentry/sentry/blob/3a86fa61bf4293c0eed9837f6b0529bda15dfbc4/src/sentry/models/apitoken.py#L135

Reply:
It just depends what you want to support in the method, and the caller has to make sure to appease the contract. What is the expected flow or contract we want to have?

Reply:
Ultimately, as we improve all of the various token types, I would like to require `AuthTokenType`. All newly generated tokens would be classified with this and we wouldn't allow a `None`. Right now, the `None` is there for backwards compatibility.

Since we've set `choices` on the `CharField`, then I suspect we will end up with a validation error if someone provides a string that doesn't match one of the choices. 🤔 I'll test that out as we progress through the token types.